### PR TITLE
Support Extended Auditing policy

### PR DESCRIPTION
**Pull request by bashahee**
## Description

This PR enables managing (setting/ getting) an `Extended Auditing` Policy of a server/ database by adding a new Parameter `PredicateExpression` to existing `Auditing` cmdlets. In addition, existing cmdlets for managing (setting/ getting) an Auditing Policy of a server/ database were modified to use SQL clients instead of Legacy SQL clients. Moreover, existing scenario tests were slightly modified, and new ones were added in order to test modified cmdlets.

Corresponding review issue was opened in https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/195

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by MiYanni** [comment]
@azuresdkci Retest this please

**Comment by MiYanni** [comment]
@bashahee You have failing SQL tests that need investigation.

**Comment by bashahee** [comment]
@MiYanni Can this PR be approved?

**Comment by MiYanni** [comment]
@azuresdkci Retest this please

**Comment by MiYanni** [comment]
@bashahee You have a build error:
```
11:30:44   Target                          Description                                                                            
11:30:44   ------                          -----------                                                                            
11:30:44   Set-AzureRmSqlServerAuditing.md Trailing yaml string found in description for parameter State. Please move the trail...
11:30:44   
11:30:44   
11:30:44   A markdown file containing the help for a cmdlet is incomplete. Please check the exceptions provided for more details.
11:30:44   At C:\workspace\powershell\tools\GenerateHelp.ps1:49 char:9
11:30:44   +         throw "A markdown file containing the help for a cmdlet is in ...
11:30:44   +         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
11:30:44       + CategoryInfo          : OperationStopped: (A markdown file...r more details.:String) [], RuntimeException
11:30:44       + FullyQualifiedErrorId : A markdown file containing the help for a cmdlet is incomplete. Please check the excepti 
11:30:44      ons provided for more details.
```

**Comment by MiYanni** [comment]
@azuresdkci Retest this please

**Comment by MiYanni** [comment]
@bashahee You seem to be missing some test recordings. 2 test failures:
```powershell
Microsoft.Azure.Commands.Sql.Test.ScenarioTests.BlobAuditingTests.TestExtendedAuditingOnDatabase
System.ArgumentException : Unable to find recorded mock file 'c:\workspace\powershell\src\ResourceManager\Sql\Commands.Sql.Test\bin\Debug\SessionRecords\Microsoft.Azure.Commands.Sql.Test.ScenarioTests.BlobAuditingTests\TestExtendedAuditingOnDatabase.json'.\r\nParameter name: callerIdentity
   at Microsoft.Azure.Test.HttpRecorder.HttpMockServer.Initialize(String callerIdentity, String testIdentity, HttpRecorderMode mode)
   at Microsoft.Rest.ClientRuntime.Azure.TestFramework.MockContext.Start(String className, String methodName)
   at Microsoft.Azure.Commands.ScenarioTest.SqlTests.SqlTestsBase.RunPowerShellTest(String[] scripts) in c:\workspace\powershell\src\ResourceManager\Sql\Commands.Sql.Test\ScenarioTests\SqlTestsBase.cs:line 80
   at Microsoft.Azure.Commands.Sql.Test.ScenarioTests.BlobAuditingTests.TestExtendedAuditingOnDatabase() in c:\workspace\powershell\src\ResourceManager\Sql\Commands.Sql.Test\ScenarioTests\BlobAuditingTests.cs:line 266

Microsoft.Azure.Commands.Sql.Test.ScenarioTests.BlobAuditingTests.TestExtendedAuditingOnServer
System.ArgumentException : Unable to find recorded mock file 'c:\workspace\powershell\src\ResourceManager\Sql\Commands.Sql.Test\bin\Debug\SessionRecords\Microsoft.Azure.Commands.Sql.Test.ScenarioTests.BlobAuditingTests\TestExtendedAuditingOnServer.json'.\r\nParameter name: callerIdentity
   at Microsoft.Azure.Test.HttpRecorder.HttpMockServer.Initialize(String callerIdentity, String testIdentity, HttpRecorderMode mode)
   at Microsoft.Rest.ClientRuntime.Azure.TestFramework.MockContext.Start(String className, String methodName)
   at Microsoft.Azure.Commands.ScenarioTest.SqlTests.SqlTestsBase.RunPowerShellTest(String[] scripts) in c:\workspace\powershell\src\ResourceManager\Sql\Commands.Sql.Test\ScenarioTests\SqlTestsBase.cs:line 80
   at Microsoft.Azure.Commands.Sql.Test.ScenarioTests.BlobAuditingTests.TestExtendedAuditingOnServer() in c:\workspace\powershell\src\ResourceManager\Sql\Commands.Sql.Test\ScenarioTests\BlobAuditingTests.cs:line 278
```

**Comment by MiYanni** [comment]
@azuresdkci Retest this please

**Comment by MiYanni** [comment]
@azuresdkci Retest this please

**Comment by MiYanni** [comment]
@bashahee You have some cred scan issues. Check the `CredScanIssues.csv` file in the build. If you find that those files use non-persistent keys or passwords, then just add those files to the suppression json.

**Comment by bashahee** [comment]
@MiYanni Where can the suppression file be found?

**Comment by MiYanni** [comment]
@azuresdkci Retest this please

**Comment by bashahee** [comment]
@Miyanni I did not add those two files since they do not exist anymore, and I updated all .csproj filea accordingly. I do not know why the build is referring them.

**Comment by MiYanni** [comment]
@bashahee Those files are still there in your fork/branch:
https://github.com/bashahee/azure-powershell/tree/preview/src/ResourceManager/Sql/Commands.Sql.Test/SessionRecords/Microsoft.Azure.Commands.Sql.Test.ScenarioTests.ExtendedAuditingTests

**Comment by MiYanni** [comment]
@azuresdkci Retest this please

File: src/ResourceManager/Sql/Commands.Sql.Test/Commands.Sql.Test.csproj [hunk 1]
```diff
@@ -59,6 +59,15 @@
...
     </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Microsoft.Rest.ClientRuntime.2.3.12\lib\net452\Microsoft.Rest.ClientRuntime.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime.Azure, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Microsoft.Rest.ClientRuntime.Azure.3.3.13\lib\net452\Microsoft.Rest.ClientRuntime.Azure.dll</HintPath>
+    </Reference>
+    <Reference Include="System.Net.Http.WebRequest" />
+    <Reference Include="System.Runtime" />
+    <Reference Include="System.Runtime.Serialization" />
```
Comment (MiYanni) [inline src/ResourceManager/Sql/Commands.Sql.Test/Commands.Sql.Test.csproj]: These additional references should not be added. They are provided via our .targets files.

File: src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/AuditingClassicStorageTests.cs [hunk 2]
```diff
@@ -33,9 +33,11 @@ protected override void SetupManagementClients(RestTestFramework.MockContext con
...
         {
         }
-     
-        [Fact]
-        [Trait(Category.AcceptanceType, Category.CheckIn)]
+
+        // Commenting out the test until a fix will be added by yaiyun
+        // [Fact]
+        // [Trait(Category.AcceptanceType, Category.CheckIn)]
+        // See issue https://github.com/Azure/azure-powershell/issues/6601
```
Comment (MiYanni) [inline src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/AuditingClassicStorageTests.cs]: Instead:

```cs
[Fact(Skip = "Skipping the test until a fix will be added by yaiyun: https://github.com/Azure/azure-powershell/issues/6601")]
[Trait(Category.AcceptanceType, Category.CheckIn)]
```

File: src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/BlobAuditingClassicStorageTests.cs [hunk 3]
```diff
@@ -33,9 +33,11 @@ protected override void SetupManagementClients(RestTestFramework.MockContext con
...
         {
         }
-     
-        [Fact]
-        [Trait(Category.AcceptanceType, Category.CheckIn)]
+
+        // Commenting out the test until a fix will be added by yaiyun.
+        // [Fact]
+        // [Trait(Category.AcceptanceType, Category.CheckIn)]
+        // See issue https://github.com/Azure/azure-powershell/issues/6601
```
Comment (MiYanni) [inline src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/BlobAuditingClassicStorageTests.cs]: Instead:
```cs
[Fact(Skip = "Skipping the test until a fix will be added by yaiyun: https://github.com/Azure/azure-powershell/issues/6601")]
[Trait(Category.AcceptanceType, Category.CheckIn)]
```

File: src/ResourceManager/Sql/Commands.Sql.Test/packages.config [hunk 4]
```diff
@@ -17,6 +17,8 @@
   <package id="Microsoft.Bcl.Build" version="1.0.14" targetFramework="net45" />
   <package id="Microsoft.IdentityModel.Clients.ActiveDirectory" version="2.28.3" targetFramework="net45" />
   <package id="Microsoft.Net.Http" version="2.2.28" targetFramework="net45" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.12" targetFramework="net452" />
+  <package id="Microsoft.Rest.ClientRuntime.Azure" version="3.3.13" targetFramework="net452" />
```
Comment (MiYanni) [inline src/ResourceManager/Sql/Commands.Sql.Test/packages.config]: These do not need to be added.

File: src/ResourceManager/Sql/Commands.Sql/Commands.Sql.csproj [hunk 5]
```diff
@@ -499,6 +499,16 @@
...
+    <Reference Include="Microsoft.Rest.ClientRuntime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Microsoft.Rest.ClientRuntime.2.3.12\lib\net452\Microsoft.Rest.ClientRuntime.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime.Azure, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Microsoft.Rest.ClientRuntime.Azure.3.3.13\lib\net452\Microsoft.Rest.ClientRuntime.Azure.dll</HintPath>
+    </Reference>
+    <Reference Include="Newtonsoft.Json, Version=6.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Newtonsoft.Json.6.0.8\lib\net45\Newtonsoft.Json.dll</HintPath>
+    </Reference>
+    <Reference Include="System.Runtime" />
```
Comment (MiYanni) [inline src/ResourceManager/Sql/Commands.Sql/Commands.Sql.csproj]: These references should not be added.

File: src/ResourceManager/Sql/Commands.Sql/packages.config [hunk 6]
```diff
@@ -5,5 +5,8 @@
   <package id="Microsoft.Data.Edm" version="5.8.2" targetFramework="net452" />
   <package id="Microsoft.Data.OData" version="5.8.2" targetFramework="net452" />
   <package id="Microsoft.Data.Services.Client" version="5.8.2" targetFramework="net452" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.12" targetFramework="net452" />
+  <package id="Microsoft.Rest.ClientRuntime.Azure" version="3.3.13" targetFramework="net452" />
+  <package id="Newtonsoft.Json" version="6.0.8" targetFramework="net452" />
```
Comment (MiYanni) [inline src/ResourceManager/Sql/Commands.Sql/packages.config]: These packages should not be added.

File: src/ResourceManager/Sql/Commands.Sql/help/Set-AzureRmSqlDatabaseAuditing.md [hunk 7]
```diff
@@ -57,29 +58,39 @@ PS C:\>Set-AzureRmSqlDatabaseAuditing -State Disabled -ResourceGroupName "Resour
...
 SELECT on dbo.myTable by public
-SELECT on DATABASE::myDatabase by public
-SELECT on SCHEMA::mySchema by public
+SELECT on DATABASE::myDatabase by public 
+SELECT on SCHEMA::mySchema by public 
 For more information, see https://docs.microsoft.com/en-us/sql/relational-databases/security/auditing/sql-server-audit-action-groups-and-actions#database-level-audit-actions.
 
 ```yaml
-Type: System.String[]
+Type: String[]
```
Comment (MiYanni) [inline src/ResourceManager/Sql/Commands.Sql/help/Set-AzureRmSqlDatabaseAuditing.md]: You've regenerated your help files incorrectly. You need to use the `-UseFullTypeName` switch as indicated in our documentation: https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md#updating-help-after-making-cmdlet-changes

**Comment by MiYanni** [comment]
@bashahee I've reviewed the PR. A few minor issues to address.

**Comment by MiYanni** [comment]
@azuresdkci Retest this please

**Comment by MiYanni** [comment]
@bashahee You have a merge conflict.

**Comment by bashahee** [comment]
@Miyanni Can we now merge?

**Comment by MiYanni** [comment]
@bashahee You have a merge conflict.
image

**Comment by bashahee** [comment]
@MiYanni I think I managed to resolve conflicts.

**Comment by MiYanni** [comment]
@azuresdkci Retest this please

**Pull request merged**